[RF] Documentation Changes requested by @guitargeek for the "RooFit AD Support" doc

QuillPusher opened the pull request:
These changes are on top of the first draft that I submitted, that was merged as part of [PR13929](https://github.com/root-project/root/pull/13929).

The requested changes can be tracked in my personal drafts repo [here](https://github.com/QuillPusher/QP_Drafts/pull/3).

Changes:

- RooFit intro before AD support -> already covered in "Overview on RooFit implementation details..." section by @guitargeek

- **Extracting free functions** to a separate header file is **optional**, re-wrote it as one of the two currently recommended approaches

- changed to "Support **Code Generation**" instead of "AD Support" in most places

- minor formatting and spelling errors changed

# This Pull request:

## Changes or fixes:


## Checklist:

- [ ] tested changes locally
- [ ] updated the docs (if necessary)

This PR fixes #

guitargeek commented on roofit/doc/developers/roofit_ad.md:
```diff
@@ -865,24 +927,25 @@ console (useful for debugging).
 
 [EvaluateFuncs]: https://github.com/root-project/root/blob/4e8c577dfd6a19d7c38a74e3074b406a598bf76a/roofit/roofitcore/inc/RooFit/Detail/EvaluateFuncs.h
 
+[AnalyticalIntegrals]: https://github.com/root-project/root/blob/4e8c577dfd6a19d7c38a74e3074b406a598bf76a/roofit/roofitcore/inc/RooFit/Detail/AnalyticalIntegrals.h
```
```suggestion
[AnalyticalIntegrals]: https://github.com/root-project/root/blob/master/roofit/roofitcore/inc/RooFit/Detail/AnalyticalIntegrals.h
```
Instead of pointing to a random commit, can you maybe point to ROOT master? Also for all the other `root-project/root` links in this file.

guitargeek commented on roofit/doc/developers/roofit_ad.md:
```diff
@@ -92,22 +108,67 @@ On a high level, this *code generation* is implemented as follows:
+header files).
+
+- to reuse common code, both in the generated code, and in the existing 
+evaluate() method (meaning that the refactoring of `evaluate()` is optional, 
```
```suggestion
RooAbsReal::evaluate() method (meaning that the refactoring of `evaluate()` is optional, 
```
At least the first time any method is introduced in this document, I would use the full function name including the class so that Doxygen generates a clickable link.

guitargeek commented on roofit/doc/developers/roofit_ad.md:
````diff
@@ -505,7 +567,7 @@ output using the `buildCall()` function.
 ```
 
 \note The implementation of the `buildCallToAnalyticIntegral()` function is
````
Also at least the first time you mention `buildCallToAnalyticIntegral()`, you should fully qualify the name, including the class name `RooAbsReal::`.

QuillPusher commented:
Thanks @guitargeek ,

- added fully qualified function names in some places for better Doxygen rendering (I wasn't too sure about this, please review)

- Removed random commit IDs and added branch name to repo links

vgvassilev commented:
@guitargeek ping. Is there anything else to be done here?

guitargeek reviewed (approved):
Thank you very much for implementing the suggestions and addressing the comments!

I have reorganized the commit message a bit with an improved commit massage for your commit, and added some other commits with general documentation and code improvements to be merged together when the CI is green.

guitargeek commented:
@phsft-bot build

guitargeek commented:
@phsft-bot build

guitargeek commented:
@phsft-bot build
